cloud-storage-rules-runtime crashes due to missing com/ibm/icu/text/UTF16$StringComparator

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->

<!--
Thank you for contributing to the Firebase community!

Think you found a bug?
=======================
Yeah, we're definitely not perfect! Please use this template and include a minimal repro when opening the issue. If you know how to solve the issue, please create a Pull Request, and we'd be happy to review it!

Have a feature request?
========================
Great, we love hearing how we can improve our products! However, do not use this template to submit a feature request. Please submit your feature requests to: https://firebase.google.com/support/contact/bugs-features/

Have a usage question?
=======================
We get lots of those and we love helping you, but GitHub is not the best place for them and they will be closed. Please take a look at the guide first: https://firebase.google.com/docs/cli/

If the official documentation doesn't help, try asking through our official support channel: https://firebase.google.com/support/

Additional locations to check for solutions or assistance from the community:
- Stack Overflow: https://stackoverflow.com/
- Firebase Slack Community: https://firebase.community/

*Please avoid duplicate posting across multiple channels!*
-->

My apologies if this is not the right repository to file this in.
It looks like cloud-storage-rules-runtime isn't part of this repository but is being invoked by [downloadableEmulators.ts](https://github.com/firebase/firebase-tools/blob/master/src/emulator/downloadableEmulators.ts).


### [REQUIRED] Environment info

<!-- What version of the Firebase CLI (`firebase-tools`) are you using? Note that your issue may already be fixed in the latest versions. The latest version can be found at https://github.com/firebase/firebase-tools/releases -->

<!-- Output of `firebase --version` -->

**firebase-tools:** 9.12.1

<!-- e.g. macOS, Windows, Ubuntu -->

**Platform:** macOS

### [REQUIRED] Test case

Create the following files:

Create a test file (this is the easiest way I've found to reproduce this so far), test.ts:
```
import * as rulesUnitTesting from '@firebase/rules-unit-testing';

const myApp = rulesUnitTesting.initializeTestApp({
  storageBucket: 'my-bucket',
});

describe('some sample tests', () => {
  it('should throw an exception', async () => {
    await myApp.storage().ref('folder/1.txt').putString('abcd');
    await myApp.storage().ref('folder/1.txt').getMetadata();
  });
});
```

storage.rules:
```
rules_version = '2';
service firebase.storage {
  match /b/{bucket}/o {
    match /{allPaths=**} {
      allow write: if true;
      allow read: if request.auth.token.keys().hasAny(resource.metadata.keys());
    }
  }
}
```
### [REQUIRED] Steps to reproduce

1) Launch the firebase storage emulator.
2) Run the test file.

### [REQUIRED] Expected behavior

The test completes without the storage emulator crashing.

<!-- What is the expected behavior? -->

### [REQUIRED] Actual behavior

The storage emulator crashes with the following exception is thrown:
```
  some sample tests
⚠  Unexpected rules runtime output: Exception in thread "main" 
⚠  Unexpected rules runtime output: java.lang.NoClassDefFoundError: com/ibm/icu/text/UTF16$StringComparator
        at com.google.firebase.rules.runtime.common.UnicodeComparator.<clinit>(UnicodeComparator.java:20)
        at com.google.firebase.rules.runtime.impl.types.MapType$MapKeys.doExecute(MapType.java:216)
        at com.google.firebase.rules.runtime.utils.GuardedRuntimeFunction.lambda$execute$0(GuardedRuntimeFunction.java:28)
        at com.google.firebase.rules.runtime.utils.InvocationGuard.execute(InvocationGuard.java:36)
        at com.google.firebase.rules.runtime.utils.GuardedRuntimeFunction.execute(GuardedRuntimeFunction.java:27)
        at com.google.firebase.rules.runtime.impl.StackMachine.executeFunction(StackMachine.java:674)
        at com.google.firebase.rules.runtime.impl.StackMachine.callFunction(StackMachine.java:651)
        at com.google.firebase.rules.runtime.impl.StackMachine.evaluateCall(StackMachine.java:452)
        at com.google.firebase.rules.runtime.impl.StackMachine.evaluateExpression(StackMachine.java:240)
        at com.google.firebase.rules.runtime.impl.StackMachine.evaluateNextExpression(StackMachine.java:209)

⚠  Unexpected rules runtime output:     at com.google.firebase.rules.runtime.impl.StackMachine.evaluate(StackMachine.java:166)
        at com.google.firebase.rules.runtime.impl.IterativeInterpreter.evaluate(IterativeInterpreter.java:33)
        at com.google.firebase.rules.runtime.impl.Interpreter.interpret(Interpreter.java:28)
        at com.google.firebase.rules.runtime.impl.DefaultEvaluator.evaluate(DefaultEvaluator.java:130)
        at com.google.firebase.rules.tools.local.server.EmulatorRuleClient$EmulatorRuleEvaluator.evaluate(EmulatorRuleClient.java:76)
        at com.google.firebase.rules.tools.local.server.EmulatorRuleClient$EmulatorRuleEvaluator.evaluate(EmulatorRuleClient.java:64)
        at com.google.firebase.rules.tools.local.server.ServerActionVerify.perform(ServerActionVerify.java:107)
        at com.google.firebase.rules.tools.local.server.Server.performActionFromLine(Server.java:79)
        at com.google.firebase.rules.tools.local.server.Server.handleRequest(Server.java:46)
        at com.google.firebase.rules.tools.local.server.Server.start(Server.java:36)
        at com.google.firebase.rules.tools.local.FirebaseRulesTooling.main(FirebaseRulesTooling.java:83)

⚠  Unexpected rules runtime output: Caused by: java.lang.ClassNotFoundException: com.ibm.icu.text.UTF16$StringComparator
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)

⚠  Unexpected rules runtime output:     ... 21 more
```

### Workaround:
Not really a workaround but for now I manually added com/ibm/icu/text/UTF16$StringComparator to the java class path using these steps:

1) Download the jar containing java.lang.NoClassDefFoundError: com/ibm/icu/text/UTF16$StringComparator, for example: https://mvnrepository.com/artifact/com.ibm.icu/icu4j/69.1
2) Place it in a library that's easily accessible, for example $HOME/jars/
3) Open downloadableEmulators.js from firebase-tools, it should be here: node_modules/firebase-tools/lib/emulator/downloadableEmulators.js
4) Find the entry for storage
5) Edit it so it looks like:
    storage: {
        binary: "java",
        args: [
            "-classpath",
            \`$DIRECTORY_FROM_STEP_1/icu4j-69.1.jar:${getExecPath(types_1.Emulators.STORAGE)}\`,
            "-Duser.language=en",
            "com.google.firebase.rules.tools.local.FirebaseRulesTooling",
            "serve",
        ],
...

6) Run the emulator

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,1 +1,2 @@
 - Support loading Firebase Functions packaged as an ES module. (#3485)
+- Fixes Cloud Storage emulator missing classes (#3541)

--- a/src/emulator/downloadableEmulators.ts
+++ b/src/emulator/downloadableEmulators.ts
@@ -51,16 +51,15 @@ export const DownloadDetails: { [s in DownloadableEmulators]: EmulatorDownloadDe
     },
   },
   storage: {
-    downloadPath: path.join(CACHE_DIR, "cloud-storage-rules-runtime-v1.0.0.jar"),
-    version: "1.0.0",
+    downloadPath: path.join(CACHE_DIR, "cloud-storage-rules-runtime-v1.0.1.jar"),
+    version: "1.0.1",
     opts: {
       cacheDir: CACHE_DIR,
       remoteUrl:
-        "https://storage.googleapis.com/firebase-preview-drop/emulator/cloud-storage-rules-runtime-v1.0.0.jar",
-      expectedSize: 63857175,
-      expectedChecksum: "fd8577f82d42ee1c03ae9d12b888049c",
+        "https://storage.googleapis.com/firebase-preview-drop/emulator/cloud-storage-rules-runtime-v1.0.1.jar",
+      expectedSize: 32729999,
+      expectedChecksum: "1a441f5e16c17aa7a27db71c9c9186d5",
       namePrefix: "cloud-storage-rules-emulator",
-      skipChecksumAndSize: true,
     },
   },
   ui: {

